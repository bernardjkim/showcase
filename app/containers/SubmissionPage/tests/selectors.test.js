import { fromJS } from 'immutable';
import {
  selectSubmissionPageDomain,
  makeSelectError,
  makeSelectLoading,
} from '../selectors';

const initialState = fromJS({
  loading: false,
  error: false,
});

describe('selectSubmissionPageDomain', () => {
  it('should select the submission page state', () => {
    expect(selectSubmissionPageDomain(initialState)).toMatchSnapshot();
  });
});

describe('makeSelectLoading', () => {
  const loadingSelector = makeSelectLoading();
  it('should select the loading state', () => {
    expect(loadingSelector(initialState)).toMatchSnapshot();
  });
});

describe('makeSelectError', () => {
  const errorSelector = makeSelectError();
  it('should select the error state', () => {
    expect(errorSelector(initialState)).toMatchSnapshot();
  });
});
