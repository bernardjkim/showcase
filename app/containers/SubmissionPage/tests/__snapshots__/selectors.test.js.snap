// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeSelectError should select the error state 1`] = `false`;

exports[`makeSelectLoading should select the loading state 1`] = `false`;

exports[`selectSubmissionPageDomain should select the submission page state 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
}
`;
